enhancements/storage/cinder-csi-driver-operator-configuration.md

```diff
@@ -0,0 +1,355 @@
+  <br>
+  We estimate that the false positive risk of attempting to validate all
+  possible OpenStack authentication configurations is significantly higher than
+  the likelihood of any user actually having done this. We therefore judge that
+  it is safer not to attempt to handle this situation explicitly, and simply
+  constrain the user to modify their configuration to use the cluster cloud
+  credentials. If any user had put non-credential information in the credentials
+  secret, this would also be a prompt to them to fix that.
+</dd>
+<dt>bs-version</dt>
+<dt>trust-device-path</dt>
+<dd>
+  These options are not supported by the Cinder CSI driver and will therefore
+  cause parse issues if included in configuration. These are unnecessary knobs
+  and we will simply drop them when upgrading this configuration.
+</dd>
+</dl>
+
+We will always add `use-clouds=true` and `cloud="openstack"` in the generated
+`openshift-cluster-csi-drivers/cloud-conf` config map.  This will refer to the
+cluster-managed cloud credentials. If the user specifies these values in the
+user-managed config map, they will be ignored and overridden.
+
+If the user specifies configuration that would not parse after transformation,
+we will mark OCCDO as `Upgradable=False`, and will not update the transformed
+configuration.
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+This enhancement is an essential feature for the migration to external CCM as
+the Cinder CSI driver must be used with this. We cannot continue to provide all
+existing functionality without the ability to customise the Cinder CSI
+configuration. This feature does not directly provide any new user-facing
+functionality; its purpose is to preserve existing functionality when migrating
+to the Cinder CSI driver. Therefore the primary validation of this feature will
+be that our existing test coverage does not regress when upgrading to external
+CCM and switching the Cinder CSI driver.
+
+We will ensure full coverage of edge cases which are impractical to test on real
+hardware through unit testing.
+
+#### Existing OpenStack test coverage
+
+QE already regularly runs a full test suite against a matrix of the following
+test configurations:
+
+- Networking:
+  - OpenshiftSDN
+  - OVNKubernetes
+  - Kuryr
+- Installation:
+  - IPI
+  - IPI with Proxy
+  - UPI
+- OpenStack version:
+  - OSP 16.1
+  - OSP 16.2
+
+Specifically, they test:
+
+- conformance/parallel
+- cinder-csi
+- manilla
+- openstack-test
+
+The test configurations already cover the full range of documented
+configurations changes, as can be seen in the [test code][cloud-cm-gerrit].
+
+[cloud-cm-gerrit]: https://code.engineering.redhat.com/gerrit/gitweb?p=openshift-ir-plugin.git;a=blob;f=roles/tests/tasks/change_cloud_cm.yml;h=33ee741b1ef9c45c0417b911d234b4d3d83e68e2;hb=HEAD
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature is part of Cinder CSI driver, which is already tech preview.
```

Comment:
Actually, Cinder CSI driver is already GA. It's becoming the default in 4.11.

Reply:
Done